homeassistant/components/xiaomi_miio/climate.py

```diff
@@ -0,0 +1,519 @@
+"""Support for Xiaomi Mi Air Dehumidifier."""
+import asyncio
+import logging
+from enum import Enum
+from functools import partial
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.climate import (
+    ClimateDevice,
+    PLATFORM_SCHEMA,
+    DOMAIN
+)
+from homeassistant.components.climate.const import (
+    HVAC_MODE_DRY,
+    HVAC_MODE_OFF,
+    ATTR_HVAC_MODES,
+    ATTR_CURRENT_HUMIDITY,
+    ATTR_HUMIDITY,
+    ATTR_MIN_HUMIDITY,
+    ATTR_MAX_HUMIDITY,
+    ATTR_FAN_MODE,
+    ATTR_FAN_MODES,
+    ATTR_PRESET_MODE,
+    ATTR_PRESET_MODES,
+    SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE
+)
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_HOST,
+    CONF_TOKEN,
+    ATTR_ENTITY_ID,
+    TEMP_CELSIUS
+)
+from homeassistant.exceptions import PlatformNotReady
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Xiaomi Miio Device'
+DATA_KEY = 'fan.xiaomi_miio'
+
+CONF_MODEL = 'model'
+MODEL_AIRDEHUMIDIFIER_V1 = 'nwt.derh.wdh318efw1'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_TOKEN):
+            vol.All(cv.string, vol.Length(min=32, max=32)),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MODEL): vol.In(
+            [
+                MODEL_AIRDEHUMIDIFIER_V1,
+            ]
+        ),
+    }
+)
+
+ATTR_MODEL = 'model'
+
+# Air Dehumidifier
+ATTR_TEMPERATURE = 'temperature'
+ATTR_MODE = 'mode'
+ATTR_BUZZER = 'buzzer'
+ATTR_CHILD_LOCK = 'child_lock'
+ATTR_LED = 'led'
+ATTR_FAN_SPEED = 'fan_speed'
+ATTR_TARGET_HUMIDITY = 'target_humidity'
+ATTR_TANK_FULL = 'tank_full'
+ATTR_COMPRESSOR_STATUS = 'compressor_status'
+ATTR_DEFROST_STATUS = 'defrost_status'
+ATTR_FAN_ST = 'fan_st'
+ATTR_ALARM = 'alarm'
```

Comment:
You could move these constants or even more into a separate file e.g. `const.py` and import them into this file.